src/index.ts

```diff
@@ -17,25 +19,68 @@ export const plugin: PluginFunction<{}> = async schema => {
       }
     }
   });
+  const chunks: Code[] = [];
 
-  const content = await code`
+  chunks.push(code`
     export const possibleTypes = {
       ${Object.entries(interfaceImpls).map(([name, impls]) => {
         return `${name}: [${impls.map(n => `"${n}"`).join(", ")}],`;
       })}
     };
-    
-    ${addUnionTypes(interfaceImpls)}
-  `.toStringWithImports();
-
-  // Also add type unions of the possible types for use in code if desired
+  `);
 
+  if (config.generateUnionTypes) {
+    chunks.push(...addUnionTypes(interfaceImpls, config));
+  }
 
+  const content = await code`${chunks}`.toStringWithImports();
   return { content } as PluginOutput;
 };
 
-function addUnionTypes(interfaceImpls: Record<string, string[]>) {
-  return Object.entries(interfaceImpls).map(([name, impls]) => `
-  export type ${name}Types = ${impls.join(" | ")};
-  `).join("");
+// Also add type unions of the possible types for use in code if desired
+function addUnionTypes(interfaceImpls: Record<string, string[]>, config: PossibleTypesConfig): Code[] {
+  return Object.entries(interfaceImpls).map(
+    ([name, impls]) => code`
+  export type ${name}Types = ${joinCodes(
+      impls.map(entityName => toImp(config.entityImportPattern.replace(entityNamePlaceholder, entityName))),
+      " | ",
+    )};
+  `,
+  );
+}
+
+interface Config {
+  possibleTypes?: Partial<PossibleTypesConfig>;
+}
+
+interface PossibleTypesConfig {
+  generateUnionTypes: boolean;
+  entityImportPattern: string;
+}
+
+const entityNamePlaceholder = "[ENTITY]";
+const defaultConfig: PossibleTypesConfig = {
+  generateUnionTypes: false,
+  entityImportPattern: `src/entities#${entityNamePlaceholder}`,
+};
+
+/** A `.join(...)` that doesn't `toString()` elements so that they can stay codes. */
+export function joinCodes(elements: unknown[], delimiter: string): unknown[] {
+  const result: unknown[] = [delimiter];
```

Comment:
I'm surprised the delimiter starts out as the first entry in the list?

Reply:
Yea, me too. Borrowed that from the simple resolvers. Maybe prettier takes it out 🤷